File: MAP/JavaRepo - Toy language/lab2/src/model/Statement/CompStmt.java
package model.Statement;

import utils.*;
import utils.adt.MyIStack;

public class CompStmt implements IStmt {
    private IStmt first;
    private IStmt snd;

    public CompStmt(IStmt f1, IStmt f2)
    {
        first = f1;
        snd =f2;
    }

    @Override
    public String toString(){
        return "(" + first + ";" + snd + ")";
    }

    @Override
    public PrgState execute(PrgState state){
        MyIStack<IStmt> stk = state.getExeStack();
        stk.push(snd);
        stk.push(first);
        //return state;

        return null;
    }
}

File: MAP/C# - Toy language/Lab10/Toy Language/Toy Language/model/statement/CompStmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language.utils;

namespace Toy_Language.model.statement
{
    public class CompStmt : IStmt
    {
        private IStmt first;
        private IStmt second;

        public CompStmt(
            IStmt _first,
            IStmt _second
            )
        {
            first = _first;
            second = _second;
        }

        public PrgState Execute(PrgState state)
        {
            IExeStack<IStmt> exeStack = state.ExeStack;
            exeStack.Push(first);
            exeStack.Push(second);

            return state;
        }

        public override string ToString()
        {
            return "(" + first + ";" + second + ")";
        }


    }
}
